plugins/google.js

```diff
@@ -26,35 +27,132 @@ hoverZoomPlugins.push({
             '$1'
         );
 
-        // Extract direct links to images using document.scripts data (no api call(s) needed)
-        // Mostly inspired from chocolatebot's userscript : Google DWIMages 
+        // Hook Google 'Open' XMLHttpRequests to catch data & metadata associated with pictures displayed
+        // These requests are issued by client to Google servers in order to obtain new data when user scrolls down
+        // Hooked data is stored in scripts in DOM with class='hoverZoomHookedData'
+        if ($('script.hoverZoomHook').length == 0) { // Inject hook script in document if not already there
+            var hookScript = document.createElement('script');
+            hookScript.type = 'text/javascript';
+            hookScript.text = `if (typeof oldXHROpen !== 'function') { // Hook only once!
+                console.log('Hooking XHROpen');
+                oldXHROpen = window.XMLHttpRequest.prototype.open;
+                window.XMLHttpRequest.prototype.open = function(method, url, async, user, password) {
+                    // catch responses
+                    this.addEventListener('load', function() {
+                        try {
+                            //console.log('load: ' + this.responseText);
+                            // store response as plain text in a <script> element for later usage by plug-in
+                            var hookedDataScript = document.createElement('script');
+                            hookedDataScript.type = 'text';
+                            hookedDataScript.text = this.responseText;
+                            hookedDataScript.classList.add('hoverZoomHookedData'); // add specific class for easier retrieval
+                            (document.head || document.documentElement).appendChild(hookedDataScript);
+                            // Add & remove empty <a> element to/from DOM to trigger HoverZoom,
```

Comment:
Instead of doing that hack, why not call appropriate hoverzoom function directly?

Reply:
Yes, it's a hack to trigger the function: prepareImgLinksAsync(dontResetDelay)
I can change it, but this function would have to become public ?

Reply:
I think calling prepareImgLinksAsync would be better. It's ok to make it public.

Reply:
Unfortunatly, the code that intercepts XHR responses is executed <b>in the page context</b>, not in the extension context. So i did not manage to call prepareImgLinksAsync from here, maybe you have a solution?

Reply:
Hm, ok, let's keep it as-is for now and maybe we'll come up with a less hacky approach.